protocol/messenger.go

```diff
@@ -216,7 +216,31 @@ type mailserverCycle struct {
 	allMailservers            []mailserversDB.Mailserver
 	activeMailserver          *mailserversDB.Mailserver
 	peers                     map[string]peerStatus
-	availabilitySubscriptions []chan struct{}
+	availabilitySubscriptions *availabilitySubscriptions
+}
+
+type availabilitySubscriptions struct {
+	sync.Mutex
+	subscriptions []chan struct{}
+}
+
+func (s *availabilitySubscriptions) Subscribe() <-chan struct{} {
+	s.Lock()
+	defer s.Unlock()
+	c := make(chan struct{})
+	s.subscriptions = append(s.subscriptions, c)
+	return c
+}
+
+func (s *availabilitySubscriptions) EmitMailserverAvailable() {
+	s.Lock()
+	defer s.Unlock()
+
+	for _, subs := range s.subscriptions {
+		close(subs)
+		l := len(s.subscriptions)
+		s.subscriptions = s.subscriptions[:l-1]
+	}
```

Comment:
This will effectively reset the list, right? Maybe make it more clear?
```suggestion
	for _, subs := range s.subscriptions {
		close(subs)
	}
	s.subscriptions = make([]chan struct{}, 0)
```